x-pack/filebeat/input/entityanalytics/internal/collections/set.go

```diff
@@ -0,0 +1,101 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package collections
+
+import "encoding/json"
+
+// Set is a collection data structure that retains single values of a given type.
+type Set[T comparable] struct {
```

Comment:
That remains to be seen. Next integration we'll be working on will be Okta, so we'll see if this is needed or not. If I had to guess, `uuid.UUID` and `string` would the two most common types used. If it were limited to that, I wouldn't be opposed to de-generalizing Set and Tree into UUID and string implementations. It would be easier to address your concerns with sorting above.

Reply:
Implementing a `compar` on a `[16]byte` is fairly straightforward. I'd do that.

Reply:
I'm still looking into doing this, but I don't want to have it hold up the review if it takes too long. I was trying to be clever and use something like this:

```go
type UUIDSet map[uuid.UUID]struct{}
```

I ran into a number of issues trying to use that with a custom JSON unmarshaler. It doesn't work well unless you mix pointer and value receivers, which I'm sure the linter will hate. I'll probably just stick with the method I'm currently using.